src/microsoft/office/license/user/microsoft-office-license-user.controller.js

```diff
@@ -1,73 +1,50 @@
+        this.translator = translator;
     }
 
     $onInit () {
-        this.currentLicense = this.$stateParams.serviceName;
-
-        this.usersDetails = [];
-        this.loaders = {
-            users: false,
-            userPoll: {}
-        };
-
-        this.$scope.$on("microsoft.office.license.user.add", this.delayedGetUser);
-        this.$scope.$on("microsoft.office.license.user.edit", this.delayedGetUser);
-        this.$scope.$on("microsoft.office.license.user.delete", this.delayedGetUser);
+        this.$scope.$on("microsoft.office.license.user.add", () => this.delayedGetUsers());
+        this.$scope.$on("microsoft.office.license.user.edit", () => this.delayedGetUsers());
+        this.$scope.$on("microsoft.office.license.user.delete", () => this.delayedGetUsers());
 
-        this.getUsersId();
+        this.getUserIds();
     }
 
-    transformItem (item) {
-        return this.license.getUserDetails(this.$scope.currentLicense, item)
+    transformItem ({ id }) {
+        return this.license.getUserDetails(this.$scope.currentLicense, id)
             .then((details) => {
                 if (details.status !== "ok") {
-                    this.loaders.userPoll[item] = true;
-
-                    this.license.pollUserDetails(this.$scope.currentLicense, item, this.$scope)
-                        .then(() => {
-                            this.loaders.userPoll.item = false;
-                            return this.delayedGetUser();
-                        })
-                        .catch((err) => {
-                            this.loaders.userPoll.item = false;
-                            return err;
-                        });
+                    details.isLoading = true;
+                    this.license.pollUserDetails(this.$scope.currentLicense, id, this.$scope)
+                        .then(() => this.delayedGetUsers())
+                        .finally(() => { details.isLoading = false; });
                 }
                 return details;
             })
             .catch((err) => {
-                this.loaders.userPoll.item = false;
-                return err;
+                this.Alerter.error(this.translator.tr("microsoft_office_license_detail_user_error") + err.message, this.$scope.alerts.main);
+                return { id, activationEmail: id };
             });
     }
 
-    onTransformItemDone () {
-        this.loaders.users = false;
-    }
-
 
-    getUsersId () {
-        this.loaders.users = true;
+    getUserIds () {
         this.users = null;
 
         return this.license.getUsers(this.currentLicense)
-            .then((users) => { this.users = users; })
-            .catch((err) => this.Alerter.error(err.message, this.$scope.alerts.tabs))
-            .finally(() => {
-                if (_.isEmpty(this.users)) {
-                    this.loaders.users = false;
-                }
-            });
+            .then((users) => { this.users = users.map((id) => ({ id })); })
```

Comment:
Please use Lodash instead of ES6 for native functions :smiley:

Reply:
Hello, I think we need to have a discussion on it because, there's two think schools of thought. Some who think we should always use lodash and some who think we should use ES6 when lodash doesn't allow us to win in lisibility or time.

Reply:
I stand for a large use of our native language, but yes we should have a clear point of view for effective for all small teams on that.

Reply:
@FredericEspiau I am curious on why you insist on the usage of Lodash. Now, to be fair, I was the first to advocate and insist on the use of lodash last year. ,The context evolved: we did not have es6 support across all stacks then. .map, .find, or simple checks such as Array.isArray were not widely supported across our browser targets. So I have to revise that position.

I still think we need to use lodash when it implements algorithms that we'd have to implement ourselves (less effort and less maintenance) and/or when it improves readability. But that does not mean we need to avoid the native es6 functionalities when available, quite the contrary.

So I'd like to hear the opposing arguments to get a full picture. Code style convention for uniformity?

Thanks!

Reply:
@FredericEspiau still interested in your opinion on the topic.